packages/harpguru-core/src/components/harp-guru/hooks/use-flush-buffered-activity-toggles/use-flush-buffered-activity-toggles.ts

```diff
@@ -0,0 +1,48 @@
+import { useGlobal } from 'reactn'
+import { unstable_batchedUpdates } from 'react-native'
+import { useEffect } from 'react'
+import type { HarpStrataProps } from 'harpstrata'
+import { getHarpStrata } from 'harpstrata'
+
+import { batchToggleDegreeIds } from '../../utils'
+
+export const useFlushBufferedActivityToggles = (): void => {
+  const [bufferedActivityToggles, setBufferedActivityToggles] = useGlobal(
+    'bufferedActivityToggles'
+  )
+  const [activeHarpStrata, setActiveHarpStrata] = useGlobal('activeHarpStrata')
+
+  useEffect(() => {
+    const flushBufferedToggles = setTimeout(() => {
+      if (bufferedActivityToggles.length === 0) return
+      const {
+        apparatus: { id: apparatusId },
+        pozitionId,
+        harpKeyId,
+        isActiveComplex: { activeDegreeIds },
+      } = activeHarpStrata
+      const newActiveDegreeIds = batchToggleDegreeIds(
+        activeDegreeIds,
+        bufferedActivityToggles
+      )
+      const newHarpStrataProps: HarpStrataProps = {
+        apparatusId,
+        pozitionId,
+        harpKeyId,
+        activeIds: newActiveDegreeIds,
+      }
+      unstable_batchedUpdates(() => {
+        setActiveHarpStrata(getHarpStrata(newHarpStrataProps))
+        setBufferedActivityToggles([])
+      })
+    }, 200)
+    return () => {
+      clearTimeout(flushBufferedToggles)
+    }
+  }, [
+    bufferedActivityToggles,
+    setBufferedActivityToggles,
+    setActiveHarpStrata,
+    activeHarpStrata,
```

Comment:
I'm quite sure that the only dependency we need here is the `bufferedActivityToggles`. The rest are in just to be safe. We can consider removing them as part of an optimisation step. Logged in #34

Reply:
Logged in #34 as planned: https://github.com/js-jslog/harpguru/issues/34#issuecomment-715900172